### ISSUE TITLE
Issues with header based column selection

### ISSUE THREAD

**Issue by learnbyexample** [sstadick/hck#26]
1) Giving a filename isn't working. Passing `stdin` works.

```bash
$ printf 'a,b,c\n1,2,3\n' > test.csv

# hangs
$ hck -d, -F "a" test.csv
^C

# stdin works
$ printf 'a,b,c\n1,2,3\n' | hck -d, -F "a"
a
1
```

2) If `-F` isn't the last option when giving filename, it works, but the header gets printed twice.

```bash
$ hck -F "a" -F "b" -d, test.csv
a	b
a	b
1	2

# stdin works as expected again
$ printf 'a,b,c\n1,2,3\n' | hck -F "a" -F "b" -d,
a	b
1	2
```

3) Last field header isn't being recognized. Works if input is single line without newline.

```bash
$ printf 'a,b,c\n1,2,3\n' | hck -d, -F "c"
[2021-07-15T10:39:57Z ERROR hck] No headers matched

$ printf 'a,b,c' | hck -d, -F "c"
c
```

4) There's no error if there are unmatched headers, but at least one header matched. If this is preferred as the default behavior, adding an option to get error would be useful.

```bash
$ printf 'a,b,c\n1,2,3\n' | hck -d, -F "a" -F "xyz"
a
1
```

**Comment by sstadick** [comment]
Fixes incoming for all of the above! https://github.com/sstadick/hck/pull/27

Thank you for the awesome bug report 👍

**Comment by sstadick** [comment]
See release v0.5.2 which should be uploaded shortly.

**Comment by sstadick** [comment]
Okay, the v0.5.3 release should have all the fixes. v0.5.2 would raise an error if you excluded a header and it couldn't find that header, which I classified as a bug and fixed with v0.5.3.

**Comment by learnbyexample** [comment]
Thanks for the quick fix :)
